Faillible directions

ManevilleF opened the pull request:
I have a use case where calling `H3Cell::direction` panics with an invalid direction of 7.
This means that there should be a faillible API for directions because it can fail on resolution 0.

I also improved some code a bit and removed redundant clippy configuration I added a while ago

nmandery commented:
Good point with making this API fallible as well, the recent move to H3 v4 made most API methods return Result types as well. Interesting finding with the invalid direction, I am curious whats the cause for this.

Thank you!

nmandery commented:
That seems strange - I would have expected basecells to have no direction.

ManevilleF commented:
That's why I removed the test checking that 0 res cells have directions and replaced it with a test on how they instead fail.
I also added this case in the error and panic documentation for the directions
